### ISSUE TITLE
SCons cant clean read-only files

### ISSUE THREAD

**Issue by bdbaddog** [SCons/scons#3026]
This issue was originally created at: 2016-04-21 00:09:59.
This issue was reported by: `flow86`.
flow86 said at 2016-04-21 00:09:59
>Hi,
> 
>scons cant "clean" readonly files. Since you can "install" files which are read
>only then, you can't clean them anymore and get an error.
> 
>In the below patch the "CleanTask" adds a chmod before deletion
> 
>```
>Index: SCons/Script/Main.py
>===================================================================
>--- SCons/Script/Main.py	(revision 10149)
>+++ SCons/Script/Main.py	(revision 10150)
>@@ -306,6 +306,10 @@
>     def fs_delete(self, path, pathstr, remove=True):
>         try:
>             if os.path.lexists(path):
>+                try:
>+                    os.chmod( path, 0777 )
>+                except:
>+                    pass
>                 if os.path.isfile(path) or os.path.islink(path):
>                     if remove: os.unlink(path)
>                     display("Removed " + pathstr)
>@@ -315,6 +319,9 @@
>                         p = os.path.join(path, e)
>                         s = os.path.join(pathstr, e)
>                         if os.path.isfile(p):
>+                            try:
>+                                os.chmod(p, 0777)
>+                            except: pass
>                             if remove: os.unlink(p)
>                             display("Removed " + s)
>                         else:
>```

bdbaddog said at 2017-08-20 19:20:25
>I'm not sure this is advisable.
>It's clearly o.k. when SCons has created and chmod'd the file originally, but it could have been 
>chmod'd outside of SCons' control.
> 
>Thoughts?

**Comment by mwichmann** [comment]
Is this the same issue as #2117?

**Comment by bdbaddog** [comment]
Yes. Duplicate.  Though I guess the title of 2117 should remove win32 because it's all platforms right?

**Comment by mwichmann** [comment]
I believe so, yes. There doesn't seem to be any actual mention of Windows other than the title.

**Comment by mwichmann** [comment]
Later information suggests this is a windows problem - windows can't remove or replace read-only files; Linux looks at the directory permissions for that.
